nemo_rl/environments/genrm_environment.py

```diff
@@ -0,0 +1,277 @@
+    EnvironmentReturn,
+)
+
+class GenRMEnvironmentMetadata(TypedDict):
+    num_responses: int
+    helpfulness_1: Optional[int]
+    helpfulness_2: Optional[int]
+    preference_ranking: Optional[int]
+
+
+@ray.remote
+class GenRMEnvironment(EnvironmentInterface):
+    """Generative Reward Model environment for HelpSteer3 dataset."""
+    
+    def __init__(self, cfg: dict):
+        self.cfg = cfg
+        self.format_penalty = cfg.get("format_penalty", -100)  # Large penalty for format violations
+        logging.basicConfig(level=logging.INFO)
+    
+    def extract_answer(self, string: str) -> Optional[str]:
+        """Extract Answer String from \\boxed expression."""
+        # Try to find the last occurrence of \boxed
+        idx = string.rfind("\\boxed")
+        if idx < 0:
+            # Try \fbox as alternative
+            idx = string.rfind("\\fbox")
+            if idx < 0:
+                return None
+
+        # Find the matching braces
+        i = idx + 6  # Skip past "\boxed"
+        if i >= len(string) or string[i] != '{':
+            return None
+            
+        # Count braces to find the matching closing brace
+        brace_count = 0
+        start_idx = i
+        while i < len(string):
+            if string[i] == '{':
+                brace_count += 1
+            elif string[i] == '}':
+                brace_count -= 1
+                if brace_count == 0:
+                    # Extract content between braces
+                    return string[start_idx + 1:i].strip()
+            i += 1
+
+        return None
+
+    def parse_score_value(self, score_str: str) -> Optional[int]:
+        """Parse a score value from a string, handling various formats."""
+        if not score_str:
+            return None
+            
+        score_str = score_str.strip()
+        
+        # Try direct conversion first
+        try:
+            return int(score_str)
+        except ValueError:
+            pass
+        
+        # Try to extract the first number found
+        match = re.search(r'(\d+)', score_str)
+        if match:
+            try:
+                return int(match.group(1))
+            except ValueError:
+                pass
+                
+        return None
+
+    def distance_abs(self, predicted: Any, ground_truth: Any) -> int:
+        """Calculate absolute distance between predicted and ground truth."""
+        try:
+            # Handle None values
+            if predicted is None or ground_truth is None:
+                return 100
+            
+            # Convert to integers if they're strings
+            if isinstance(predicted, str):
+                predicted = self.parse_score_value(predicted)
+                if predicted is None:
+                    return 100
+                    
+            if isinstance(ground_truth, str):
+                ground_truth = self.parse_score_value(ground_truth)
+                if ground_truth is None:
+                    return 100
```

Comment:
Small simplification:
```suggestion
            if isinstance(predicted, str):
                predicted = self.parse_score_value(predicted)
                
            if isinstance(ground_truth, str):
                ground_truth = self.parse_score_value(ground_truth)

            # Handle None values
            if predicted is None or ground_truth is None:
                return 100
```